src/test/javascript/spec/component/learning-paths/components/learning-paths-table.component.spec.ts

```diff
@@ -0,0 +1,147 @@
+        fixture.detectChanges();
+
+        const learningPathRows = fixture.nativeElement.querySelectorAll('tr');
+
+        expect(getLearningPathInformationSpy).toHaveBeenCalledExactlyOnceWith(courseId, pageable);
+
+        expect(component.learningPaths()).toEqual(searchResults.resultsOnPage);
+        expect(learningPathRows).toHaveLength(searchResults.resultsOnPage.length + 1);
+        expect(component.collectionSize()).toBe(searchResults.resultsOnPage.length);
+    });
+
+    it('should open competency graph modal', async () => {
+        const learningPathId = 1;
+        const openCompetencyGraphSpy = jest.spyOn(component, 'openCompetencyGraph');
+
+        fixture.detectChanges();
+        await fixture.whenStable();
+        fixture.detectChanges();
+
+        const learningPathIdButton = fixture.debugElement.query(By.css(`#open-competency-graph-button-${learningPathId}`));
+        learningPathIdButton.nativeElement.click();
+        expect(openCompetencyGraphSpy).toHaveBeenCalledExactlyOnceWith(1);
+    });
+
+    it('should change page', async () => {
+        const onPageChangeSpy = jest.spyOn(component, 'setPage');
+
+        fixture.detectChanges();
+        await fixture.whenStable();
+
+        await component.setPage(2);
+
+        expect(onPageChangeSpy).toHaveBeenLastCalledWith(2);
+        expect(getLearningPathInformationSpy).toHaveBeenLastCalledWith(courseId, { ...pageable, page: 2 });
+    });
+
+    it('should search for learning paths when the search term changes', async () => {
+        fixture.detectChanges();
+        await fixture.whenStable();
+        fixture.detectChanges();
+
+        const searchField = fixture.debugElement.query(By.css('#learning-path-search'));
+        const searchPageable = { ...pageable, searchTerm: 'Search Term' };
+        searchField.nativeElement.value = 'Search Term';
+        searchField.nativeElement.dispatchEvent(new Event('input'));
+
+        fixture.detectChanges();
+        await fixture.whenStable();
+
+        expect(getLearningPathInformationSpy).toHaveBeenLastCalledWith(courseId, searchPageable);
+    });
+
+    it('should show error message when loading learning paths fails', async () => {
+        getLearningPathInformationSpy.mockRejectedValue(new Error('Error loading learning paths'));
+        const alertServiceErrorSpy = jest.spyOn(alertService, 'addAlert');
+
+        fixture.detectChanges();
+        await fixture.whenStable();
+
+        expect(alertServiceErrorSpy).toHaveBeenCalledOnce();
+    });
+
+    it('should set isLoading correctly', async () => {
+        const isLoadingSpy = jest.spyOn(component.isLoading, 'set');
+
+        fixture.detectChanges();
+        await fixture.whenStable();
+
+        expect(isLoadingSpy).toHaveBeenNthCalledWith(1, true);
+        expect(isLoadingSpy).toHaveBeenNthCalledWith(2, false);
+    });
+
+    function generateResults(start: number = 0, end: number): LearningPathInformationDTO[] {
```

Comment:
I'll leave it up to you, but as it's currently used, we don't really need the default value for `start`

```suggestion
    function generateResults(start: number, end: number): LearningPathInformationDTO[] {
```